src/utils/defaultMenuRenderer.js

```diff
@@ -18,7 +18,7 @@ function menuRenderer ({
 	let Option = optionComponent;
 
 	return options.map((option, i) => {
-		let isSelected = valueArray && valueArray.indexOf(option) > -1;
+		let isSelected = valueArray && valueArray.findIndex(x => x[valueKey] == option[valueKey]) > -1;
```

Comment:
Instead of including a polyfill what about just using the `some` method from Array?